modules/kafka/tests/test_kafka_config.c

```diff
@@ -0,0 +1,163 @@
+  log_pipe_unref(&driver->super);
+}
+
+Test(kafka_config, test_bootstrap_server_is_mandatory)
+{
+  LogDriver *driver = kafka_dd_new(configuration);
+  _setup_topic(driver, "test-topic");
+  cr_assert_not(kafka_dd_init(&driver->super));
+  assert_grabbed_log_contains("kafka: the bootstrap-servers() option is required for kafka destinations");
+
+  log_pipe_deinit(&driver->super);
+  log_pipe_unref(&driver->super);
+}
+
+Test(kafka_config, test_mandatory_options)
+{
+  LogDriver *driver = kafka_dd_new(configuration);
+  _setup_topic(driver, "default-test-topic");
+  _setup_bootstrap_server(driver, "test-host:9092");
+  cr_assert(kafka_dd_init(&driver->super));
+
+  log_pipe_deinit(&driver->super);
+  log_pipe_unref(&driver->super);
+}
+
+static void
+_setup_kafka_property(LogDriver *driver, const gchar *name, const gchar *value)
+{
+  KafkaDestDriver *self = (KafkaDestDriver *)driver;
+  KafkaProperty *prop = kafka_property_new(name, value);
+  GList *property_list = g_list_prepend(NULL, prop);
+  kafka_dd_merge_config(driver, property_list);
+}
+
+static void
+_assert_configured_property(KafkaDestDriver *kafka_driver, const gchar *property_name, const gchar *expected_value)
+{
+  const rd_kafka_conf_t *kafka_conf = rd_kafka_conf(kafka_driver->kafka);
+  cr_assert(kafka_conf != NULL);
+
+  gchar configured_property[1024];
+  size_t prop_size = 0;
+  rd_kafka_conf_get(kafka_conf, property_name, configured_property, &prop_size);
+  cr_assert_str_eq(configured_property, expected_value);
+}
+
+Test(kafka_config, cannot_override_protected_property)
+{
+  LogDriver *driver = kafka_dd_new(configuration);
+  _setup_topic(driver, "default-test-topic");
+  const gchar *bootstrap_servers_option = "test-host:9092";
+  _setup_bootstrap_server(driver, bootstrap_servers_option);
+  _setup_kafka_property(driver, "bootstrap.servers", "host2:9092");
+  cr_assert(kafka_dd_init(&driver->super));
+
+  _assert_configured_property((KafkaDestDriver *) driver, "bootstrap.servers", bootstrap_servers_option);
+
+  log_pipe_deinit(&driver->super);
+  log_pipe_unref(&driver->super);
+}
+
+Test(kafka_config, protected_property_alias)
+{
+  LogDriver *driver = kafka_dd_new(configuration);
+  _setup_topic(driver, "default-test-topic");
+  const gchar *bootstrap_servers_option = "test-host:9092";
+  _setup_bootstrap_server(driver, bootstrap_servers_option);
+  _setup_kafka_property(driver, "metadata.broker.list", "host2:9092");
+  cr_assert(kafka_dd_init(&driver->super));
+
+  _assert_configured_property((KafkaDestDriver *) driver, "bootstrap.servers", bootstrap_servers_option);
+
+  log_pipe_deinit(&driver->super);
+  log_pipe_unref(&driver->super);
+}
+
+Test(kafka_config, property_setting_works)
+{
+  LogDriver *driver = kafka_dd_new(configuration);
+  _setup_topic(driver, "default-test-topic");
+  _setup_bootstrap_server(driver, "test-host:9092");
+  _setup_kafka_property(driver, "retries", "1000");
+  _setup_kafka_property(driver, "queue.buffering.max.messages", "65535");
```

Comment:
Could you also please check with `topic.partitioner`?

Reply:
Good idea to test if we can override a property that is set by default in KafkaDestDriver.